### PR TITLE
Add createTicketAsync method.

### PR DESCRIPTION
**Pull request by johnou**
Fixes https://github.com/cloudbees/zendesk-java-client/issues/211

### CODE REVIEW

File: src/main/java/org/zendesk/client/v2/Zendesk.java [hunk 1]
```diff
@@ -273,6 +273,12 @@ public void deleteTicket(long id) {
...
     }
 
+    public ListenableFuture<JobStatus<Ticket>> createTicketAsync(Ticket ticket) {
+        return submit(req("POST", cnst("/tickets.json?async=true"),
+                JSON, json(Collections.singletonMap("ticket", ticket))),
+                handleJobStatus(Ticket.class));
+    }
+
     public Ticket createTicket(Ticket ticket) {
         return complete(submit(req("POST", cnst("/tickets.json"),
```
Comment (bsideup) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: shouldn't it use `complete(createTicketAsync(ticket))` now? :)

File: src/main/java/org/zendesk/client/v2/Zendesk.java [hunk 2]
```diff
@@ -273,6 +273,12 @@ public void deleteTicket(long id) {
         complete(submit(req("DELETE", tmpl("/tickets/{id}.json").set("id", id)), handleStatus()));
     }
 
+    public ListenableFuture<JobStatus<Ticket>> createTicketAsync(Ticket ticket) {
+        return submit(req("POST", cnst("/tickets.json?async=true"),
```
Comment (bsideup) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: Is it possible to create a ticket without `async=true` but with a `ListenableFuture` as a return type?

Comment (johnou) [reply src/main/java/org/zendesk/client/v2/Zendesk.java]: If you can come up with a good name for it so we can have both i'll put it right in ;)

Comment (bsideup) [reply src/main/java/org/zendesk/client/v2/Zendesk.java]: `ListenableFuture<JobStatus<Ticket>> createTicketAsync(Ticket ticket, boolean sendToBackground)` ? :)

Comment (johnou) [reply src/main/java/org/zendesk/client/v2/Zendesk.java]: one returns a job status, the other returns an actual ticket https://developer.zendesk.com/rest_api/docs/core/tickets#create-ticket

Comment (bsideup) [reply src/main/java/org/zendesk/client/v2/Zendesk.java]: `ListenableFuture<Ticket> createTicketAsync(Ticket ticket)`
`ListenableFuture<JobStatus<Ticket>> createTicketDeferred(Ticket ticket)`

? :)

Comment (johnou) [reply src/main/java/org/zendesk/client/v2/Zendesk.java]: I think that works I'll update the pr when I get back to the keyboard.

**Comment by johnou** [comment]
@stephenc PTAL, I'm not completely happy with the method names as the deferred method uses a param called async..

File: src/main/java/org/zendesk/client/v2/Zendesk.java [hunk 3]
```diff
@@ -273,10 +273,20 @@ public void deleteTicket(long id) {
         complete(submit(req("DELETE", tmpl("/tickets/{id}.json").set("id", id)), handleStatus()));
     }
 
+    public ListenableFuture<JobStatus<Ticket>> queueCreateTicketAsync(Ticket ticket) {
```
Comment (johnou) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: @bsideup ptal, using the same terminology Zendesk use in their documentation.

Comment (bsideup) [reply src/main/java/org/zendesk/client/v2/Zendesk.java]: 👍

**Comment by johnou** [comment]
@aheritier PTAL

**Comment by stephenc** [comment]
FTR I am fine with having `Async` on methods that return a future as long as we are consistent

**Pull request merged**